storage_service/locations/api/resources.py

```diff
@@ -769,3 +773,176 @@ def _attempt_package_request_event(self, package, request_info, event_type, even
+        }
+
+        for f in bundle.obj.file_set.all():
+            response["files"].append({attr: getattr(f, attr) for attr in ('source_id', 'name', 'source_package', 'checksum', 'accessionid', 'origin')})
+
+        return http.HttpResponse(status=200, content=json.dumps(response),
+            content_type='application/json')
+
+    def file_data(self, request, **kwargs):
+        """
+        Returns file metadata as a JSON array of objects.
+
+        This maps properties of the File class to the names of the
+        Elasticsearch indices' Transferfile index, allowing this to directly
+        substitute for Elasticsearch when looking up metadata on specific files.
+
+        Acceptable parameters are:
+            * relative_path (searches the `name` field)
+            * fileuuid (searches the `source_id` field)
+            * accessionid (searches the `accessionid` field)
+            * sipuuid (searches the `source_package` field)
+
+        :returns: an array of one or more objects. See the transferfile
+        index for information on the return format.
+        If no results are found for the specified query, returns 404.
+        If no acceptable query parameters are found, returns 400.
+        """
+        property_map = {
+            "relative_path": "name",
+            "fileuuid": "source_id",
+            "accessionid": "accessionid",
+            "sipuuid": "source_package"
+        }
+        query = {}
+        for source, dest in property_map.iteritems():
+            try:
+                query[dest] = request.GET[source]
+            except KeyError:
+                pass
+
+        if not query:
+            response = {
+                "success": False,
+                "error": "No supported query properties found!"
+            }
+            return http.HttpBadRequest(content=json.dumps(response),
+                content_type="application/json")
+
+        files = File.objects.filter(**query)
+        if not files.exists():
+            return http.HttpNotFound()
+
+        response = []
+        for f in files:
+            response.append({
+                "accessionid": f.accessionid,
+                "file_extension": os.path.splitext(f.name)[1],
+                "filename": os.path.basename(f.name),
+                "relative_path": f.name,
+                "fileuuid": f.source_id,
+                "origin": f.origin,
+                "sipuuid": f.source_package
+            })
+
+        return http.HttpResponse(content=json.dumps(response), content_type="application/json")
+
+    @_custom_endpoint(expected_methods=['delete'])
+    def remove_files_from_package(self, request, bundle, **kwargs):
+        """
+        Removes all file records associated with this package.
+        """
+
+        bundle.obj.file_set.all().delete()
+        return http.HttpNoContent()
+
+    @_custom_endpoint(expected_methods=['post'])
+    def add_files_to_package(self, request, bundle, **kwargs):
```

Comment:
Rather than having add_files and remove_files, why not have file/<uuid>/files/ that responds to DELETE or PUT? Seems more RESTful.